backend/typescript/services/implementations/reviewService.ts

```diff
@@ -32,6 +37,156 @@ class ReviewService implements IReviewService {
     if (db !== sequelize) sequelize.close(); // Using test db instead of main db
   }
 
+  /* eslint-disable class-methods-use-this, no-await-in-loop */
+  async findOrCreateTag(
+    review: PgReview,
+    tag: TagRequest,
+    t: Transaction,
+  ): Promise<PgTag> {
+    const newTag = await PgTag.findOrCreate({
+      where: { name: tag.name },
+      transaction: t,
+    }).then((data) => data[0]);
+    await review.$add("tags", newTag, { transaction: t });
+    return newTag;
+  }
+
+  async findOrCreateTags(
+    review: PgReview,
+    tags: TagRequest[],
+    t: Transaction,
+  ): Promise<TagResponse[]> {
+    const tagsRet: TagResponse[] = [];
+    for (let i = 0; i < tags.length; i += 1) {
+      const tag = await this.findOrCreateTag(review, tags[i], t);
+      tagsRet.push({ id: tag.id, name: tag.name });
+    }
+    return tagsRet;
+  }
+
+  async findOrCreateSeries(
+    seriesName: string | null | undefined,
```

Comment:
why can this be null or undefined?

Reply:
To allow the frontend to either not define the field or pass it with a null value.